src/WebKit/Enumerators.cs

```diff
@@ -24,23 +26,25 @@ internal class IndexedContainerEnumerator<T> : IEnumerator<T> {
 
 		public T Current {
 			get {
-				return _container [_index];
+				return _container! [_index];
 			}
 		}
 
-		object IEnumerator.Current {
+		object? IEnumerator.Current {
 			get { return ((IEnumerator<T>) this).Current; }
 		}
 
 		public bool MoveNext () {
+			if (_container is null)
+				return false;
 			return ++_index < _container.Count;
 		}
 
 		public void Reset () {
 			_index = -1;
 		}
 
-		IIndexedContainer<T> _container;
+		IIndexedContainer<T>? _container;
```

Comment:
Why is this needed, if the only constructor takes a non-nullable `container` value and assigns it to this field?

Reply:
@rolfbjarne we need a way to init that. The compiler is complaining because the default value of a IIndexedContainer<T> is null. We could probably make the _container a private Init property with the type and remove the need of ? and remove all the Dipose code. Should make things nicer.

Reply:
@rolfbjarne @mandel-macaque Should I do this?

Reply:
@tj-devel709 yes, keep this since nulling it the field in the Dispose method might be important